lib/index.js

```diff
@@ -17,46 +17,19 @@
 ////////////////////////////////////////////////////////////////////////////
 
 const utils = require("./utils");
+const analytics = require("./submit-analytics");
 
-// Prevent React Native packager from seeing modules required with this
-const nodeRequire = require;
-
-function getRealmConstructor(environment) {
-  switch (environment) {
-    case "node.js":
-    case "electron":
-      var analytics = nodeRequire("./submit-analytics");
-      analytics.submitProductionAnalytics("Run").catch(() => {
-        // fail silently -- the caller is not responsible for handling
-        // errors in analytics submission
-      });
-
-      return nodeRequire("bindings")("realm.node").Realm;
-    case "reactnative":
-      //switch how babel transpiled code creates children objects.
-      //Inheriting from Realm.Object with class syntax does not support using Reflect.construct the way babel transpiles it.
-      //Defining Reflect.construct.sham makes the transpiled code use different standard mechanism for inheriting. (Function.apply with setPrototypeOf)
-      if (typeof Reflect !== "undefined" && Reflect.construct) {
-        Reflect.construct.sham = 1;
-      }
-      return global.Realm;
-    case "jscore":
-      return global.Realm;
-    case "chromedebugger":
-    case "vscodedebugger":
-      // This condition is for stripping "browser" folder from production bundles.
-      if (global.__DEV__) {
-        return require("./browser").default; // (exported as ES6 module)
-      } else {
-        throw new Error("Can´t use debugger if __DEV__ isn´t true.");
-      }
-    default:
-      throw new Error("Unexpected execution environment (" + environment + ")");
-  }
+const environment = utils.getEnvironment();
+if (environment !== "node.js" || environment !== "electron") {
```

Comment:
Can "reactnative" or "jscore" no longer be our environment here?

Reply:
Right. With [the `react-native` main field in the `package.json`](https://github.com/realm/realm-js/blob/hermes/package.json#L34), this code will no longer be hit on React Native: